torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -897,13 +912,15 @@ def run(
                     "stream": stream,
                 },
             ):
+                validate_args(launcher, *args, **kwargs, 2)
                 return launcher(
                     *args,
                     **kwargs,
                     grid=grid,
                     stream=stream,
                 )
         else:
+            validate_args(launcher, *args, **kwargs, 2)
             return launcher(
```

Comment:
I dont think we should be calling this on the hot path, at least with inspect.signature. Should we only call this on the first run ?